src/hidpp20.h

```diff
@@ -317,6 +317,34 @@ union hidpp20_button_binding {
 } __attribute__((packed));
 _Static_assert(sizeof(union hidpp20_button_binding) == 4, "Invalid size");
 
+struct hidpp20_color {
+	uint8_t red;
+	uint8_t green;
+	uint8_t blue;
+};
+struct hidpp20_internal_led {
+	uint8_t mode; /* 00 - off; 01 - on; 03 - cycle; 0a - breath */
+	struct hidpp20_color color;
+	union {
+		struct hidpp20_led_cycle {
+			uint16_t pad_1;
+			uint16_t rate;  /* 16000 - 20000Hz */
+			uint8_t brightness; /* 0 - 100 percent */
+			uint16_t pad_2;
+		} __attribute__((packed)) cycle;
+		struct hidpp20_led_breath {
+			uint16_t rate;
+			uint8_t pad_1;
+			uint8_t brightness;
+			uint16_t pad_2;
+			uint8_t pad_3;
```

Comment:
swap those two around please, for padding you should always go up, i.e. use uint8 to get to 16 bit, then uint16 to get to 32 bit, etc. It probably doesn't matter (and I've been tainted by the X protocol where we can't rely on packed) but it's still good form.